### PR TITLE
R4R : sync tokenFlow(IRISXP-575 )

### PR DESCRIPTION
**Pull request by wade-liwei**

### CODE REVIEW

File: service/handler/token_flow.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+				result = append(result, txOp)
+			}
+		}
+	}
+	return result
+}
+
+func parseTagsAndTx(tx document.CommonTx, tagStr string) (document.TokenFlow, bool) {
+	var result document.TokenFlow
+	fromToValue := strings.Split(tagStr, "::")
```
Comment (shineabel) [inline service/handler/token_flow.go]: fromToValue ->flowItem

File: service/handler/token_flow.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import (
+	"strings"
+
+	"github.com/irisnet/irishub-sync/store/document"
+	itypes "github.com/irisnet/irishub-sync/types"
+	"gopkg.in/mgo.v2/bson"
+	"gopkg.in/mgo.v2/txn"
+)
+
+func handleTokenFlow(blockWithtags document.Block, tx document.CommonTx) []txn.Op {
```
Comment (shineabel) [inline service/handler/token_flow.go]: blockWithtags -> blockWithTags

File: service/handler/token_flow.go [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+			}
+		}
+	}
+	return result
+}
+
+func parseTagsAndTx(tx document.CommonTx, tagStr string) (document.TokenFlow, bool) {
+	var result document.TokenFlow
+	fromToValue := strings.Split(tagStr, "::")
+	if len(fromToValue) != 6 {
```
Comment (shineabel) [inline service/handler/token_flow.go]: remove magic number, define const

File: store/document/token_flow.go [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+	TxHash      string     `bson:"tx_hash"`
+	From        string     `bson:"from"`
+	To          string     `bson:"to"`
+	Amount      store.Coin `bson:"amount"`
+	Fee         store.Fee  `bson:"fee"`
+	TxInitiator string     `bson:"tx_initiator"`
+	TxType      string     `bson:"tx_type"`
+	FlowType    string     `bson:"flow_type"`
+	Status      string     `bson:"status"`
+	Timestamp   string     `bson:"timestamp"`
```
Comment (zhiqiang-bianjie) [inline store/document/token_flow.go]: Field timestap's type should be Time

File: service/handler/tx.go [hunk 5]
```diff
@@ -34,7 +34,10 @@ func HandleTx(block *types.Block) error {
 			}
 			batch = append(batch, txOp)
 		}
+		//TODO: return txOps and save it
 		handleProposal(tx)
+		txOps := handleTokenFlow(blockWithTags, tx)
```
Comment (zhiqiang-bianjie) [inline service/handler/tx.go]: Pass the array pointer into the handleTokenFlow function, adding array elements by function, which eliminates unnecessary intermediate variables

**Pull request merged**